AmplifyPlugins/Core/AWSPluginsCore/AWSPluginOptions.swift

```diff
@@ -9,15 +9,53 @@ import Foundation
 import Amplify
 
 /// Plugin specific options type
-public struct AWSPluginOptions {
+///
+/// - Warning: Although this has `public` access, it is intended for internal use and should not be used directly
+///   by host applications. The behavior of this may change without warning.
+///   // OR SPI
+public struct AWSDataStorePluginOptions {
```

Comment:
Thoughts on the naming of this? It still feels a bit off, I would expect this type to be used with DataStore (even though DataStore provides no types that take in this type, nor `Any` type. I think something more specific to AWSAPIPlugin would be nice, like `AWSAPIPluginOptionsForDataStore` / `AWSAPIPluginDataStoreOptions` ?

Reply:
we should make a new file for this struct as well